### ISSUE TITLE
Allow Optional Fields in meds-evaluation Schema

### ISSUE THREAD

**Issue by Oufattole** [kamilest/meds-evaluation#13]
## Current Limitation
The `meds-evaluation` schema [currently defined here](https://github.com/kamilest/meds-evaluation/blob/920864840a4df3e7e09c4bc8f5c53fef53b426b5/src/meds_evaluation/evaluate.py#L188C5-L188C33) may fail if users define additional fields. This limits extensibility, particularly for use cases requiring additional metadata storage like embeddings and model outputs. The schema should also be publiclly importable, and not in a function so others can import and extend it more easily.

## Proposal
1. Make the schema importable for other repositories to import and extend
2. Allow arbitrary (or specific) optional fields in the schema definition to enable users to do more analysis and still run this evaluation suite.

### Specific Fields to Add
These fields would be particularly useful for model evaluation and analysis:

```python
# New optional fields
schema_extensions = [
    ("embeddings", pa.list_(pa.float64())),        # Learned representations for binary classification
                                                   # Useful for linear probing evaluations and producing plots
    
    ("logits", pa.list_(pa.float64())),           # Binary classification logits
    
    ("logits_sequence", pa.list_(pa.list_(pa.float64()))),  # Sequence model autoregressive 
                                                            # next-token prediction logits
    
    ("loss", pa.float64()),                       # Training/classification loss value
]
```

## Reference Implementation
A working example of schema extension can be found in [meds-torch](https://github.com/Oufattole/meds-torch/blob/31e4e543b208f19dd8edbdac34cccbf0e770a413/src/meds_torch/schemas/predict_schema.py), which successfully extends the base evaluation label schema.

**Comment by kamilest** [comment]
Thanks @Oufattole! I created a PR #14 allowing for extra fields, but I didn't specify what those fields should be as they're currently not used in MEDS-Evaluation. I also stopped using `pyarrow` as in my opinion it only made things more complicated converting stuff between `pyarrow` and `polars`.

Feel free to take a look and review if you'd like.
